plugins/modules/edpm_derive_pci_passthrough_devicespec.py

```diff
@@ -0,0 +1,650 @@
+#!/usr/bin/python3
+
+#
+# Copyright 2024 Red Hat Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+__metaclass__ = type
+
+import json
+import os
+import re
+import yaml
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.0',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: edpm_nova_derive_pci_passthrough_devicespec
+author:
+    - OpenStack EDPM Contributors
+version_added: '1.0'
+short_description: Derive pci passthrough devspec for sriov devices
+notes: []
+description:
+  - This module updates the pci device_spec addr list for nova,
+    when sriov nic partitioning is configured in EDPM nodes.
+    See openstack user guide for details of PCI passthrough formats.
+options:
+  sriov_device_spec_map:
+    description:
+      - List of SR-IOV device_spec
+    type: list
+    required: true
+    default: ''
```

Comment:
@vcandapp  used required but default is empty. mutually exclusive and get internal error

Reply:
Done